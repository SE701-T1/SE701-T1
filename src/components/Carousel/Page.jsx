import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@mui/material';
import styles from './Page.module.css';

export default function Page({ title, description }) {
    return (
        <>
            <Typography className={styles.title} variant='h5'>{title}</Typography>
            <Typography className={styles.description} variant='subtitle2'>{description}</Typography>
        </>
    )
}

Page.propTypes = {
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
};