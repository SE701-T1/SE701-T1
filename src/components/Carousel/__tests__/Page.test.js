import Page from '../Page';
import renderer, { act } from 'react-test-renderer';
const { render, screen } = require("@testing-library/react");

describe('Carousel Page', () => {

    /**
     * This tests whether the document has rendered the required tests to the screen.
     */
    it('is correctly rendered to the screen', () => {

        /*
        This invovation (prior to the expect statements) makes sure that the components are
        rendered before the assertions are made.
        */
        act(() => {
            render(<Page title='title_content' description='description_content'/>)
        });

        expect(screen.getByText('title_content')).toBeInTheDocument();
        expect(screen.getByText('description_content')).toBeInTheDocument();
    });

    /**
     * This test is similar to the one above, but this also checks if the classnames of the
     * component (styling) is as expected.
     */
    it('produces the correct DOM tree with rendered', () => {
        const tree = renderer.create(
            <Page title='title_content' description='description_content'/>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});