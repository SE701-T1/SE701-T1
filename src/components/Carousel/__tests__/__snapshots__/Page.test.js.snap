// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Page produces the correct DOM tree with rendered 1`] = `
<div>
  <h5
    className="MuiTypography-root MuiTypography-h5 title css-ag7rrr-MuiTypography-root"
  >
    title_content
  </h5>
  <h6
    className="MuiTypography-root MuiTypography-subtitle2 description css-dd2x5a-MuiTypography-root"
  >
    description_content
  </h6>
</div>
`;
